src/System.Private.CoreLib/src/System/Utf8String.Manipulation.cs

```diff
@@ -21,7 +19,7 @@ private Utf8String InternalSubstring(int startIndex, int length)
             Debug.Assert(length >= 0, "Length cannot be negative.");
             Debug.Assert(startIndex + length <= this.Length, "StartIndex and Length cannot point beyond the end of the string.");
 
-            Debug.Assert(startIndex != 0 && startIndex != this.Length, "Caller should handle StartIndex boundary conditions.");
+            Debug.Assert(startIndex != this.Length, "Caller should handle StartIndex boundary conditions.");
```

Comment:
>startIndex != this.Length [](start = 25, length = 25)

We already have the assert 
Debug.Assert(startIndex <= this.Length,

It looks strange we assert the startIndex can be same as the length and then we assert later it cannot. I would just combine 2 cases into one assert.

Reply:
Spoke offline with Tarek. Turns out we can just delete the line entirely because all of the conditions are already handled.